backbone/src/test/java/org/researchstack/backbone/ui/task/TaskViewModelTest.kt

```diff
@@ -0,0 +1,419 @@
+
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito
+import org.researchstack.backbone.step.Step
+import org.researchstack.backbone.task.Task
+import org.junit.rules.TestRule
+import org.junit.Rule
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import junit.framework.TestCase.assertEquals
+import org.junit.Assert
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.junit.MockitoJUnitRunner
+import org.researchstack.backbone.result.StepResult
+import org.researchstack.backbone.result.TaskResult
+
+@RunWith(MockitoJUnitRunner::class)
+class TaskViewModelTest {
+
+    @Rule
+    @JvmField
+    val rule: TestRule = InstantTaskExecutorRule()
+
+    @Mock
+    private var application: Application? = null
+    @Mock
+    private var intent: Intent? = null
+    @Mock
+    private var task: Task? = null
+
+    private lateinit var taskViewModel: TaskViewModel
+
+    @Before
+    fun before() {
+        whenever(intent?.getSerializableExtra(TaskActivity.EXTRA_TASK)).thenReturn(eq(task))
+    }
+
+    @Test
+    fun testInitViewModel_TaskShouldNotBeEmpty_EditModeMustBeFalse() {
+        // Assemble
+        // Act
+        taskViewModel = createViewModel()
+
+        // Assert
+        assertEquals(taskViewModel.editing, false)
+        assertEquals(taskViewModel.task, task)
+    }
+
+    @Test
+    fun testEdit_editShouldBeTrue_EditStepShouldBeUpdated_CancelMenuShouldBeTrueToHideCancelInTitleBar() {
+        // Assemble
+        val selectedEditStep = createCurrentStep("selectedEditStep")
+        taskViewModel = createViewModel()
+
+        // Act
+        taskViewModel.edit(selectedEditStep)
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, selectedEditStep)
+        Assert.assertEquals(taskViewModel.editStep.value, selectedEditStep)
+        Assert.assertEquals(taskViewModel.editing, true)
+        Assert.assertEquals(taskViewModel.hideMenuItemCancel.value, true)
+    }
+
+
+    @Test
+    fun testRemoveUpdatedLayout_editShouldBeFalse_EditStepShouldBeUpdated_CancelMenuShouldBeFalseToShowCancelInTitleBar() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+        val currentStepMocked = createCurrentStep()
+        val nextStepMocked = createCurrentStep("nextStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+        `when`(taskViewModel.isReviewStep(nextStepMocked)).thenReturn(true)
+
+        // Act
+        taskViewModel.removeUpdatedLayout()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.editing, false)
+        Assert.assertEquals(taskViewModel.updateCancelEditInLayout.value, true)
+        Assert.assertEquals(taskViewModel.hideMenuItemCancel.value, false)
+    }
+
+    @Test
+    fun testNext_ShouldUpdateCurrentStepToNextStep_shouldUpdateCurrentStepEventToNextStep() {
```

Comment:
Do you mind updating the test name to: `testNextStep_shouldSetCurrentStepToNext_andShouldUpdateCurrentStepEventToNextStep`. 

I'm interested in the first bit, because `testNext` means nothing to the reviewer who is unfamiliar with this test or how ReviewStep works.

Reply:
Ok